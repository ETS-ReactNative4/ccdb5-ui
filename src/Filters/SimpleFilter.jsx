import React from 'react'
import { connect } from 'react-redux'
import AggregationItem from './AggregationItem'
import SimpleCollapsibleFilter from './SimpleCollapsibleFilter'
import MoreOrLess from './MoreOrLess'
import './Aggregation.less'

export class SimpleFilter extends React.Component {
  componentWillReceiveProps(nextProps) {
    this.setState({
      showChildren: nextProps.showChildren
    });
  }

  render() {
    const listComponentProps = {
      fieldName: this.props.fieldName
    }

    return (
      <SimpleCollapsibleFilter title={this.props.title}
                               desc={this.props.desc}
                               showChildren={this.props.showChildren}
                               className="aggregation">
         <MoreOrLess listComponent={AggregationItem}
                     listComponentProps={listComponentProps}
                     options={this.props.options}
         />
      </SimpleCollapsibleFilter>
    )
  }
}

export const mapStateToProps = (state, ownProps) => {
  // Find all query filters that refer to the field name
  const activeChildren = state.query[ownProps.fieldName] || []

  return {
    options: state.aggs[ownProps.fieldName] || [],
    showChildren: activeChildren.length > 0
  }
}

export default connect(mapStateToProps)(SimpleFilter)
