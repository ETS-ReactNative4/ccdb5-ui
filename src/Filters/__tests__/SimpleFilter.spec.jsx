import React from 'react';
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { IntlProvider } from 'react-intl';
import ReduxSimpleFilter from '../SimpleFilter';
import { SimpleFilter, mapDispatchToProps } from '../SimpleFilter';
import renderer from 'react-test-renderer';

function setupSnapshot(initialAggs) {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query: {},
    aggs: {
      company_response: initialAggs,
      timely: 'yes'
    }
  })

  return renderer.create(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ReduxSimpleFilter fieldName="company_response"/>
      </IntlProvider>
    </Provider>
  )
}

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot([
      {key: 'foo', doc_count: 99}
    ]);
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('component:SimpleFilter', () =>{
  let target;
  let props;
  beforeEach(() => {
    props = {
      params: {
        timely: 'yes'
      }
    }

   target = new SimpleFilter(props);
  });

  describe('componentWillReceiveProps', () => {
    it('changes props correctly', () => {
      props.params.timely = 'no'
      const expected = 'no'

      target.componentWillReceiveProps(props)

      expect(target.props.params.timely).toEqual(expected)
    })
  })
})
