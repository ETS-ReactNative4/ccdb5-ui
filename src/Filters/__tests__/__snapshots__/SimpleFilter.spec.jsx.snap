exports[`initial state renders without crashing 1`] = `
<section
  className="collapsible-filter aggregation">
  <div
    className="layout-row">
    <h5
      className="flex-all" />
    <div
      className="flex-fixed toggle">
      <button
        className="a-btn a-btn__link hover"
        onClick={[Function]}>
        Show
        <span
          className="cf-icon cf-icon-plus-round" />
      </button>
    </div>
  </div>
  <p />
</section>
`;
